sentinel-core/src/main/java/com/alibaba/csp/sentinel/log/ConsoleHandler.java

```diff
@@ -29,35 +31,72 @@
  *
  * @author cdfive
  */
-class ConsoleHandler extends StreamHandler {
+class ConsoleHandler extends Handler {
 
-    public ConsoleHandler() {
-        super();
-        setOutputStream(System.out);
-    }
+    /**
+     * A Handler which publishes log records to System.out.
+     */
+    private StreamHandler stdoutHandler;
 
     /**
-     * Publish a <tt>LogRecord</tt>.
-     * <p>
-     * The logging request was made initially to a <tt>Logger</tt> object,
-     * which initialized the <tt>LogRecord</tt> and forwarded it here.
-     * <p>
-     * @param  record  description of the log event. A null record is
-     *                 silently ignored and is not published
+     * A Handler which publishes log records to System.err.
      */
+    private StreamHandler stderrHandler;
+
+    /**
+     * Current Handler according to the current publish log level.
+     */
+    private StreamHandler currentHandler;
+
+    public ConsoleHandler() {
+        this.stdoutHandler = new StreamHandler(System.out, new CspFormatter());
+        this.stderrHandler = new StreamHandler(System.err, new CspFormatter());
+        this.currentHandler = this.stdoutHandler;
+    }
+
+    @Override
+    public synchronized void setFormatter(Formatter newFormatter) throws SecurityException {
+        this.stdoutHandler.setFormatter(newFormatter);
+        this.stderrHandler.setFormatter(newFormatter);
+    }
+
+    @Override
+    public synchronized void setEncoding(String encoding) throws SecurityException, UnsupportedEncodingException {
+        this.stdoutHandler.setEncoding(encoding);
+        this.stderrHandler.setEncoding(encoding);
+    }
+
     @Override
     public void publish(LogRecord record) {
-        super.publish(record);
-        flush();
+        if (record.getLevel().intValue() >= Level.WARNING.intValue()) {
+            currentHandler = stderrHandler;
+        } else {
+            currentHandler = stdoutHandler;
+        }
+
+        currentHandler.publish(record);
+        currentHandler.flush();
```

Comment:
And performance maybe down if we flush during every publishing.

Reply:
If not call `flush()` here, the console maynot get the log info immediately, as we call log.info()
to log oneline message, we expected to view them immediately.
Start dashboard can verify it.
Besides, the `java.util.logging.ConsoleHandler` class also call flush() when publish(),
so I think its okay.